File: nrepeatedElements.py
class Solution:
    def repeatedNTimes(self, nums: List[int]) -> int:
        n = len(nums)//2
        numCount = {}
        for i in nums:
            if i in numCount.keys():
                numCount[i] += 1
            else:
                numCount[i] = 1
        
        for key in numCount.keys():
            if numCount[key] == n:
                return key

File: nrepeatedElements.cpp
class Solution {
public:
    int repeatedNTimes(vector<int>& nums) {
        int n = nums.size()/2;
        std::unordered_map<int, int> numCount;

        for (int i = 0; i<nums.size(); i++){
            auto element = numCount.find(nums[i]);
            if (element != numCount.end()){
               return nums[i]; 
            }
            else{
                
                //not found 
                numCount[nums[i]] = 1 ;
            }
        }
        return 0 ;
    }
};
